homeassistant/components/bluetooth/passive_update_processor.py

```diff
@@ -35,22 +57,224 @@ class PassiveBluetoothEntityKey:
 
 _T = TypeVar("_T")
 
+STORAGE_KEY = "bluetooth.passive_update_processor"
+STORAGE_SAVE_INTERVAL = timedelta(minutes=15)
+PASSIVE_UPDATE_PROCESSOR = "passive_update_processor"
+
+
+@dataclasses.dataclass(slots=True, frozen=False)
+class PassiveBluetoothProcessorData:
+    """Data for the passive bluetooth processor."""
+
+    coordinators: set[PassiveBluetoothProcessorCoordinator] = dataclasses.field(
+        default_factory=set
+    )
+    all_restore_data: dict[
+        str, dict[str, RestoredPassiveBluetoothDataUpdate]
+    ] = dataclasses.field(default_factory=dict)
+
+
+class RestoredPassiveBluetoothDataUpdate(TypedDict):
+    """Restored PassiveBluetoothDataUpdate."""
+
+    devices: dict[str, DeviceInfo]
+    entity_descriptions: dict[str, dict[str, Any]]
+    entity_names: dict[str, str | None]
+    entity_data: dict[str, Any]
+
+
+def serialize_passive_bluetooth_entity_key(
+    entity_key: PassiveBluetoothEntityKey,
+) -> str:
+    """Serialize a passive bluetooth entity key."""
+    return f"{entity_key.key}___{entity_key.device_id or ''}"
+
+
+def deserialize_passive_bluetooth_entity_key(
+    seralized_entity_key: str,
+) -> PassiveBluetoothEntityKey:
+    """Deserialize a passive bluetooth entity key."""
+    key, device_id = seralized_entity_key.split("___")
+    return PassiveBluetoothEntityKey(key, device_id or None)
+
+
+# Fields do not change so we can cache the result
+# of calling fields() on the dataclass
+cached_fields = cache(dataclasses.fields)
+
+
+def deserialize_entity_description(
```

Comment:
We could add as_dict and from_dict methods to the entity descriptions but this works for now. If we do it in more places that would be a good idea